src/span_ext.rs

```diff
@@ -237,4 +299,33 @@ impl OpenTelemetrySpanExt for tracing::Span {
             }
         });
     }
+
+    fn add_otel_span_event<T>(&self, name: T, attributes: Vec<KeyValue>)
+    where
+        T: Into<Cow<'static, str>>,
+    {
+        self.add_otel_span_event_with_timestamp(name, time::now(), attributes);
+    }
+
+    fn add_otel_span_event_with_timestamp<T>(
+        &self,
+        name: T,
+        timestamp: SystemTime,
+        attributes: Vec<KeyValue>,
+    ) where
+        T: Into<Cow<'static, str>>,
+    {
+        self.with_subscriber(move |(id, subscriber)| {
+            let mut event = Some(opentelemetry::trace::Event::new(
+                name, timestamp, attributes, 0,
+            ));
+            if let Some(get_context) = subscriber.downcast_ref::<WithContext>() {
+                get_context.with_context(subscriber, id, move |data, _tracer| {
+                    if let Some(event) = event.take() {
+                        data.builder.events.get_or_insert_with(Vec::new).push(event);
+                    }
+                });
+            }
```

Comment:
Nit: I'm aware this is mostly copy/pasting, but please use let-else to reduce rightward drift where possible. (If you want to fix up the original function too, that would be nice, but please keep it in a separate commit.)